mongo/job_store.go

```diff
@@ -38,13 +40,37 @@ func (m *JobStore) CreateJob(ctx context.Context, id string) (job models.Job, er
 		return models.Job{}, ErrEmptyIDProvided
 	}
 
-	// Create a Job that's populated with default values of all its attributes
-	newJob := models.NewJob(id)
-
 	s := m.Session.Copy()
 	defer s.Close()
 	var jobToFind models.Job
 
+	// Get all the jobs where state is "in-progress" and order them by "-reindex_started"
+	// (in reverse order of reindex_started so that the one started most recently is first in the Iter).
+	iter := s.DB(m.Database).C(m.Collection).Find(bson.M{"state": "in-progress"}).Sort("-reindex_started").Iter()
+	result := models.Job{}
+
+	// Check that there are no jobs in progress, which started between a calculated "check from" time and now.
+	// The checkFromTime is calculated by subtracting the configured variable "MaxReindexJobRuntime" from now.
+	checkFromTime := time.Now().Add(-1 * m.cfg.MaxReindexJobRuntime)
+	var jobStartTime time.Time
+	for iter.Next(&result) {
+		//if the start time of the job in progress is later than the checkFromTime but earlier than now then a new job should not be created yet.
+		jobStartTime = result.ReindexStarted
+		if jobStartTime.After(checkFromTime) && jobStartTime.Before(time.Now()) {
+			log.Event(ctx, "found job in progress", log.Data{"id": result.ID, "state": result.State, "start time": jobStartTime})
+			return models.Job{}, ErrExistingJobInProgress
+		}
+	}
+	defer func() {
+		err := iter.Close()
```

Comment:
This could be rewritten as one line `if err:= iter.Close; err != ni...`

Reply:
nice - done